filesystem/flysystem/FlysystemAssetStore.php

```diff
@@ -436,4 +735,91 @@ protected function getFileID($filename, $hash, $variant = null) {
 		return $fileID;
 	}
 
+	/**
+	 * Ensure each adapter re-generates its own server configuration files
+	 */
+	public static function flush() {
+		// Ensure that this instance is constructed on flush, thus forcing
+		// bootstrapping of necessary .htaccess / web.config files
+		$instance = singleton('AssetStore');
```

Comment:
I'm not clear when this flush method is called. Is it definitely called **after** the config has been re-generated? Just making sure that if you change AssetStore then flush, you don't get the old value here (which could be null and allow public access to everything).